### PR TITLE
feat: Display Spaceforce.mil news articles below the news carousel

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-771

## Proposed changes
- Display latest 4 stories from the spaceforce.mil RSS news feed
- Provide a link to the Spaceforce.mil news page below the articles

---

## Reviewer notes
Note: this PR is built off of [SC-589](https://github.com/USSF-ORBIT/ussf-portal-client/pull/744), so please review that PR first

Another note: I updated an e2e test in this repo, but also updated it in the `ussf-portal` repo and created a [PR there](https://github.com/USSF-ORBIT/ussf-portal/pull/142). Please review that PR in tandem with this one.

Third and final note (hopefully): This branch also includes the (very few) updates needed to finish [SC-772](https://app.shortcut.com/orbit-truss/story/772/finish-news-announcements-page) so please give that story a glance.

Navigate to `/news-announcements`
- Verify that only the latest four news articles from the RSS feed are being displayed
- Verify that there is a button beneath the articles with the text `View older Spaceforce.mil articles` and that when clicked, takes you to `https://www.spaceforce.mil/News` in a new tab

## Setup
Make sure to have both the CMS and the portal client running on your local machine. You can do this by
- Running `yarn services:up && yarn dev` in the CMS repo and
- Running `yarn dev` in the portal client with this branch checked out

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

### CODE REVIEW

File: src/pages/news-announcements.tsx
Comment (abbyoung) [inline src/pages/news-announcements.tsx:27]: If we only want to display 4 on the page and don't need the rest, we can change `&max=12` to `&max=4`

File: src/pages/news-announcements.tsx
Comment (abbyoung) [inline src/pages/news-announcements.tsx:89]: if the RSS_URL above has a max of 4 then we can skip this

File: src/pages/news-announcements.tsx
Comment (jcbcapps) [reply src/pages/news-announcements.tsx:27]: Awesome! I was looking for a solution like this but completely passed this line over. Thanks!

**Comment by jbecker01** [comment]
@jcbcapps this all looks good. The only thing I believe that needs to change is the CTA label to "View Details"

**Review by gidjin — APPROVED** [review]
This PR is dependent on #744 correct? If so it looks good to me, though I think you may have some updates to apply once #744 is merged. I'm marking it approved assuming that those updates happen :)

Note: I saw John's comment about `View Detail` but the Figma designs for #744 show `View article` which is what is on the other branch iirc.

**Comment by jbecker01** [comment]
@gidjin @jcbcapps you can disregard my last comment on updating the CTA to View Details. I don't think I saw that it was already updated on a different branch.

**Review by abbyoung — APPROVED** [review]

**Pull request merged**
